conda/common/logic.py

```diff
@@ -396,9 +623,28 @@ def LinearBound_(self, equation, lo, hi, preprocess, polarity):
         return res
 
     def LinearBound(self, equation, lo, hi, preprocess=True, polarity=None, name=None):
-        return self.Eval_(self.LinearBound_, (equation, lo, hi, preprocess),
-                          polarity, name, conv=False)
+        if log.isEnabledFor(DEBUG):
+            old_clause_count = self.get_clause_count()
+        ret = self.Eval_(
+            self.LinearBound_, (equation, lo, hi, preprocess), polarity, name, conv=False)
+        if log.isEnabledFor(DEBUG):
+            new_clause_count = self.get_clause_count() - old_clause_count
+            log.debug(
+                'new clauses: %d (%.0f%%)',
```

Comment:
Printing out the number of new clauses made it pretty clear where the run time and memory consumption took place :smile:. If you want, we can remove those lines again.